DirectProgramming/DPC++FPGA/Tutorials/Tools/experimental/platform_designer/Readme.md

```diff
@@ -0,0 +1,274 @@
+>
+> Linux*:
+> - For system wide installations: `/opt/intel/oneapi/setvars.sh`
+> - For private installations: `~/intel/oneapi/setvars.sh`
+>
+> Windows*:
+> - `C:\Program Files(x86)\Intel\oneAPI\setvars.bat`
+> - For PowerShell*, use the following command: `cmd.exe "/K" '"C:\Program Files (x86)\Intel\oneAPI\setvars.bat" && powershell'`
+>
+>For more information on environment variables, see **Use the setvars Script** for [Linux or macOS](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html), or [Windows](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-windows.html).
+
+Follow these steps to compile and test the design:
+1. Compile the SYCL code to RTL using a Windows or Linux machine. 
+
+   Linux:
+
+   ```bash
+   $> cd add-oneapi
+   $> mkdir build
+   $> cd build
+   $> cmake ..
+   $> make fpga_ip_export
+   ```
+
+   Windows:
+
+   ```bash
+   > cd add-oneapi
+   > mkdir build
+   > cd build
+   > cmake -G "NMake Makefiles" ..
+   > nmake fpga_ip_export
+   ```
+
+   For more details, see the Readme in the `add-oneapi` directory.
+
+2. **From the same terminal**, launch the Intel® Quartus® Prime Pro Edition GUI, and create a new Intel® Quartus® Prime project using the 'New Project' wizard. 
+
+   Linux:
+   
+   ```
+   $> cd ../../add-quartus
+   $> quartus
+   ```
+
+   Windows:
+   
+   ```
+   > cd ../../add-quartus
+   > quartus.exe
+   ```
+
+   1. Set the project directory to be the `add-quartus` directory of this code sample.
+
+   2. Set the top-level entity to be `add` to make project management easier.
+
+      ![](Readme.md.assets/quartus_new_project.png)
+
+   4. Choose **Empty Project** when prompted to select a project type.
+
+   5. Add the source file `add.v` and `jtag.sdc` to the design when the wizard prompts you.
+
+      ![](Readme.md.assets/add-files.png)
+
+   4. Make sure you choose an appropriate device. See **Board-specific Considerations** above.
+
+3. Copy the oneAPI-generated IP to the Intel Quartus® Prime project. This design uses host pipes, which generates additional internal SYCL kernels. The `fpga_export` build target uses the `-fsycl-device-code-split=per_kernel` flag to separate these additional kernels from your kernel, but these kernels will have their own reports and associated RTL. You will therefore need to hunt for the `.prj_X` directory that contains the IP you want to use in your design.
+
+   You can identify the correct `.prj_X` folder by looking for the one that contains a `*_di_inst.v` whose interfaces match your kernel. For example, in this project, `add_xample.fpga_ip.prj_1` is the correct `.prj_x` directory, because `add_example_fpga_ip_1_di_inst.v` contains only a CSR Agent interface in addition to the clock/reset signals:
```

Comment:
Are you F#@ing kidding me?  The only way to figure out which kernel we want is to look at the interfaces?  That's absurd.

Reply:
totally agree. this issue kind of goes away once the memory team gets around to implementing native host pipes, because then you don't need to use `-fsycl-device-code-split=per_kernel` anymore for host pipes.